### PR TITLE
Add Phantom type docs

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

File: docs/docs/reference/phantom-types.md [hunk 1]
```diff
@@ -0,0 +1,184 @@
+---
+layout: doc-page
+title: "Phantom Types"
+---
+
+Phantom Types  :ghost:
```
Comment (felixmulder) [inline docs/docs/reference/phantom-types.md]: This will give you the title twice

File: docs/docs/reference/phantom-types.md [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+All these operations cannot exist on phantoms as there will not be an underlying object 
+instance at runtime. At first glace this could look like a limitation, but in fact not 
+having `asInstanceOf` will make constraints more reliable as it will not be possible to 
+downcast a phantom value to fake an evidence.
+
+If they don't live in the universe bounded by `scala.Any` and `scala.Nothing` where do 
+they live? The answer is in their own type universes bounded by their phantom `Any` and `Nothing`. 
+In fact we allow multiple phantom universes to exist.
+
+```
```
Comment (felixmulder) [inline docs/docs/reference/phantom-types.md]: The highlighting component might try to guess a language here - do ` ```none` to suppress it if that's the case.

File: docs/sidebar.yml [hunk 3]
```diff
@@ -3,6 +3,10 @@ sidebar:
       url: blog/index.html
     - title: Docs
       url: docs/index.html
+    - title: Reference
```
Comment (felixmulder) [inline docs/sidebar.yml]: Should we call this "Language Reference"? I'm guessing you're following suit on what @odersky is doing with his docs. Did we decide to change this yesterday?

Comment (nicolasstucki) [inline docs/sidebar.yml]: I will rebase on whatever the result of of the other one is.

**Comment by felixmulder** [comment]
Status of this? The main page currently 404s on linking to phantom types.

**Comment by felixmulder** [comment]
Fixed in: https://github.com/lampepfl/dotty/pull/2674

But this PR should update with the correct link on the landing page.

**Comment by nicolasstucki** [comment]
@felixmulder I updated the link and rebased. It should be ready to go in. 
@biboudis do you still agree?

**Comment by biboudis** [comment]
@nicolasstucki LGTM

**Comment by felixmulder** [comment]
@biboudis - you can merge things if you give your "LGTM", Aggy :)

**Pull request merged**